### ISSUE TITLE
Feature request: add a check ensuring the minimum spacing between keyword & arguments

### ISSUE THREAD

**Issue by Lucas-C** [MarketSquare/robotframework-robocop#688]
Example of error currently detected by Visual Studio Intellisense:
![MicrosoftTeams-image (3)](https://user-images.githubusercontent.com/925560/187410120-27dc824b-2cf4-4704-90da-1d823c434899.png)

Could `robotframework-robocop` be able to detect this issue?

**Comment by Lucas-C** [comment]
**Note:** `robot --dryrun .` does not detect this problem either

**Comment by mnojek** [comment]
Hey @Lucas-C, thanks for your interest in Robocop.
Similar request as in #687 - we do not plan to implement such features. They go beyond what a "linter tool" is.
Closing as not planned.

**Comment by Lucas-C** [comment]
😔😢
OK, your call
Just FYI, a linter like Pylint does exactly this kind of static code analysis

**Comment by Gadzy** [comment]
I second Lucas here.
White-space consistency is one of the most basic functionalities of a linter tool. How is it "beyond" scope ?

**Comment by bhirsz** [comment]
We're checking for whitespace in the same way pylint is doing and even more (I will explain why I think that way later). If user provides not enough whitespace in such case we don't have any way of recognizing if we're dealing with argument or argument name - especially given how Robot Framework is flexible and allows you to input whole sentences as keywords. 

Using pylint argument as example, this python code:
```
defmy_method
```
No linter will recognize that you actually meant:
```
def my_method
```
Something similiar happens in your code.

What can you do is to detect unrecognized keyword with other tools like sherlock - but sherlock is still in growing phase and it don't have that many useful features like Robocop Or use ``robot --dryrun`` - it's actually interesting that it didn't detect this problem because normally it should). 

And coming back to my first sentences why I think Robocop is actually going bit beyong - there are cases when we detect unrecognized syntax:
```
[Arguments] arg  # only one space here
```
If some lines starts with something that looks like Robot Framework setting but it's syntax error/ unrecognized as such, we're reporting it as not-enough-whitespace-after-* errors.

**Comment by bhirsz** [comment]
If you mostly use the same libraries and you want to prevent such cases you can hovewer create custom rules. Search for keyword calls and see if their begining match keywords from your libraries. If it does but have something extra at the end - it will mean not enough whitespace. In your case keyword starts with "Click Element" which is probably from popular SeleniumLibrary but it have extra not recognized text after.

**Comment by Lucas-C** [comment]
> If user provides not enough whitespace in such case we don't have any way of recognizing if we're dealing with argument or argument name

I disagree on this point: this tool could at least support all the standard, built-in Robot Framework keywords and the ones from `robotframework-seleniumlibrary`

> No linter will recognize that you actually meant:
> `def my_method`

Agreed, but any linter would detect an incorrect syntax and hence raise an error!

> you can hovewer create custom rules

Would you have an example on how to create a custom rules that would report this problem with a "Click Element" keyword please?

**Comment by Gadzy** [comment]
@bhirsz Thank you for this clear explanation.

> given how Robot Framework is flexible and allows you to input whole sentences as keywords

This is the difficulty I missed.

Could dealing with known keywords (both standard and those exposed by imported libraries) be technically done in a way robocop detects the kind of error aforementioned ?

>  it's actually interesting that it didn't detect this problem because normally it should

Agreed. I opened a [`robotframework` issue](https://github.com/robotframework/robotframework/issues/4448).

> Would you have an example on how to create a custom rules that would report this problem with a "Click Element" keyword please?

+1 for custom rulesexample.

**Comment by Lucas-C** [comment]
I submitted #690 as a potential implementation for this rule.
I think it would be great addition to `robocop`, but we can still use it as an external rule if you don't like it.

**Comment by mnojek** [comment]
2 out of 9 rule categories were updated in #917:
- naming
- documentation
